controllers/trainingjob/trainingjob_controller_test.go

```diff
@@ -617,6 +617,294 @@ var _ = Describe("Reconciling a TrainingJob that exists", func() {
 				})
 			})
 		})
+
+		Context("TrainingJob with debug job is `in progress` has status 'Failed'", func() {
+			var failureReason string
+			var debugJobStatus sagemaker.RuleEvaluationStatus
+
+			BeforeEach(func() {
+				expectedStatus = sagemaker.TrainingJobStatusFailed
+				failureReason = "Failure within the training job"
+				debugJobStatus = sagemaker.RuleEvaluationStatusInProgress
+
+				// Add the failure reason to the describe output
+				describeOutput := CreateDescribeOutputWithDebugJobStatus(expectedStatus, expectedSecondaryStatus, debugJobStatus)
+				describeOutput.FailureReason = ToStringPtr(failureReason)
+
+				mockSageMakerClientBuilder.
+					AddDescribeTrainingJobResponse(describeOutput)
+			})
+
+			When("!HasDeletionTimestamp", func() {
+				It("Requeues after interval", func() {
+					ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				})
+
+				It("Updates status", func() {
+					ExpectStatusToBe(trainingJob, string(expectedStatus), string(expectedSecondaryStatus))
+				})
+
+				It("Has the additional field set", func() {
+					ExpectAdditionalToContain(trainingJob, failureReason)
+				})
+
+				Context("Does not have a finalizer", func() {
+					BeforeEach(func() {
+						shouldHaveFinalizer = false
+					})
+
+					It("Adds a finalizer", func() {
+						ExpectToHaveFinalizer(trainingJob, controllers.SageMakerResourceFinalizerName)
+					})
+				})
+			})
+
+			When("HasDeletionTimestamp", func() {
+				BeforeEach(func() {
+					shouldHaveDeletionTimestamp = true
+				})
+
+				It("Deletes the training job", func() {
+					ExpectTrainingJobToBeDeleted(trainingJob)
+				})
+
+				It("Requeues after interval", func() {
+					ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				})
+			})
+		})
+
+		Context("TrainingJob with debug job is `stopping` has status 'Failed'", func() {
+			var failureReason string
+			var debugJobStatus sagemaker.RuleEvaluationStatus
+
+			BeforeEach(func() {
+				expectedStatus = sagemaker.TrainingJobStatusFailed
+				failureReason = "Failure within the training job"
+				debugJobStatus = sagemaker.RuleEvaluationStatusStopping
+
+				// Add the failure reason to the describe output
+				describeOutput := CreateDescribeOutputWithDebugJobStatus(expectedStatus, expectedSecondaryStatus, debugJobStatus)
+				describeOutput.FailureReason = ToStringPtr(failureReason)
+
+				mockSageMakerClientBuilder.
+					AddDescribeTrainingJobResponse(describeOutput)
+			})
+
+			When("!HasDeletionTimestamp", func() {
+				It("Requeues after interval", func() {
+					ExpectRequeueAfterInterval(reconcileResult, reconcileError, pollDuration)
+				})
+
+				It("Updates status", func() {
+					ExpectStatusToBe(trainingJob, string(expectedStatus), string(expectedSecondaryStatus))
+				})
```

Comment:
Why are we not testing that the `debugStatus` is correctly set?

Reply:
Added the test for it, thanks for pointing this out.